pkg/dns/service.go

```diff
@@ -231,6 +243,11 @@ func FindMatchingManagedZone(originalHost, host string, zones []v1alpha1.Managed
 		return nil, "", fmt.Errorf("no valid zone found for host: %v", originalHost)
 	}
 
+	hostParts := strings.SplitN(host, ".", 2)
```

Comment:
not clear why we added this? Doesn't seem to be used?

Reply:
The original logic was outputting a pretty unuseful error message to debug the issue (no matching managedzone found), so I reworked it a little to improve the error message.